service/controller/chart/v1/resource/release/desired.go

```diff
@@ -93,6 +99,11 @@ func (r *Resource) getConfigMapValues(ctx context.Context, cr v1alpha1.Chart) (m
 func (r *Resource) getSecretValues(ctx context.Context, cr v1alpha1.Chart) (map[string]interface{}, error) {
 	secretValues := make(map[string]interface{})
 
+	if key.IsDeleted(cr) {
+		// Return early as secret has already been deleted.
+		return secretValues, nil
+	}
```

Comment:
You said you do not want to return an error when data is not found. The error is still returned below and what we said before we do not want to do is here again.

Reply:
There is a problem with removing the error. If the chart CR is not deleted and the configmap is not found that is an error state we need to handle.

I can change it to this but I think the code is more readable as it is now.

```
if apierrors.IsNotFound(err) && !key.IsDeleted(cr) { 
```

The only way I can see not to check `key.IsDeleted` is to delay deleting the resources in app-operator until chart-operator has deleted. But I think that is overly complex and not a good solution.

Reply:
I think this is again a sign that the CRUD resource interface is not a good fit here. `GetDesiredState` is called in all cases. For creating, updating and deleting. That causes more complexity here in this case and defies the purpose of the resource as we see. We said already we should have a deeper look into it. I do not want to block this here though. Go with what makes it work now and let's iterate here.

Reply:
OK cool. Yes I take your point and I'll add a TODO issue.